### PR TITLE
feat: adapted component logic to new templates

### PR DESCRIPTION
**Pull request by jacobdotcosta**
New templates require extra information from the `release.yml` file such as existence of starter, released product or belonging to a product. This PR implements this adaptation as well as adjusting the unit tests accordingly.

### CODE REVIEW

**Comment by jacobdotcosta** [comment]
Changing back to draft until I solve the CI errors.

File: src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java [hunk 1]
```diff
@@ -209,28 +209,41 @@ private void validate(Release release) {
...
+        assertTrue(component.getDescription().contains("tested and supported"));
+        assertFalse(component.getDescription().contains("It is important that you also communicate the EOL of your product so that we can plan our future releases accordingly."));
+        assertFalse(component.getDescription().contains("- product name:"));
+        component = components.get(7);
+        assertEquals("Vert.X starter", component.getName());
         final var product = component.getProduct();
         assertNotNull(product);
-        assertEquals("Narayana", product.getName());
-        assertEquals("Currently supported Narayana version information", product.getTitle());
+        assertEquals("Vert.X", product.getName());
```
Comment (cmoulliard) [inline src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java:234]: should be `Vert.x` and `Vert.X` ;-)

**Review by cmoulliard — COMMENTED** [review]
LGTM if you rename `Vert.X` to `Vert.x`

File: src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java (hunk 1 above)
Comment (jacobdotcosta) [reply src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java:234]: I'll create an issue to follow this as this requires changes to the `release.yml` files located at the BOM repository.

Comment (jacobdotcosta) [reply src/test/java/dev/snowdrop/release/services/ReleaseFactoryTest.java]: done.

**Pull request merged**